sdk/sdk.go

```diff
@@ -207,25 +207,33 @@ func New(platformEndpoint string, opts ...Option) (*SDK, error) {
 
 func SanitizePlatformEndpoint(e string) (string, error) {
 	// check if there's a scheme, if not, add https
-	if !regexp.MustCompile(`^https?://`).MatchString(e) {
-		e = "https://" + e
+	u, err := url.ParseRequestURI(e)
+	if err != nil {
+		return "", errors.Join(fmt.Errorf("cannot parse platform endpoint [%s]", e), err)
+	}
+	if u.Host == "" {
+		// if the schema is missing add https. when the schema is missing the host is parsed as the scheme
+		newE := "https://" + e
+		u, err = url.ParseRequestURI(newE)
+		if err != nil {
+			return "", errors.Join(fmt.Errorf("cannot parse platform endpoint [%s]", newE), err)
+		}
 	}
 
-	if !regexp.MustCompile(`^(https?:\/\/)?(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(:\d+)?|(localhost)(:\d+)?)\/?$`).MatchString(e) {
-		return "", errors.New("platform endpoint is not valid")
+	if u.Host == "" {
```

Comment:
nit: can you move this inside the if above? looks like this will only be true if the above block is entered

Reply:
good call, fixed